drivers/mongodb/mongodb.go

```diff
@@ -0,0 +1,211 @@
+	if d.dbName != "" {
+		return []string{d.dbName}, nil
+	} else {
+		dbNames, err := d.client.ListDatabaseNames(d.ctx, bson.D{})
+		if err != nil {
+			return nil, err
+		}
+		return dbNames, nil
+	}
+}
+
+var skipTables = []string{"local", "admin", "config"}
+
+func contains(a string, list []string) bool {
+	for _, b := range list {
+		if b == a {
+			return true
+		}
+	}
+	return false
+}
+
+func (d *Mongodb) Analyze(s *schema.Schema) error {
+	drv, err := d.Info()
+	if err != nil {
+		return errors.WithStack(err)
+	}
+	s.Driver = drv
+
+	tables := []*schema.Table{}
+	dbNames, err := d.getDatabaseNames()
+	for _, dbName := range dbNames {
+		if contains(dbName, skipTables) {
+			continue
+		}
+		dbValue := d.client.Database(dbName)
+		colls, err := dbValue.ListCollectionSpecifications(d.ctx, bson.D{})
+		if err != nil {
+			return err
+		}
+		for _, coll := range colls {
+			colVal := dbValue.Collection(coll.Name)
+			indexes, err := d.listIndexes(colVal)
+			if err != nil {
+				return err
+			}
+			estimated, err := colVal.EstimatedDocumentCount(d.ctx)
+			if err != nil {
+				return err
+			}
+			columns, err := d.listFields(colVal)
+			if err != nil {
+				return err
+			}
+			table := &schema.Table{
+				Name:    fmt.Sprintf("%s.%s", dbName, coll.Name),
+				Type:    coll.Type,
+				Columns: columns,
+				Indexes: indexes,
+				Comment: fmt.Sprintf("Count of documents is %d", estimated),
+			}
+			tables = append(tables, table)
+		}
+	}
+	s.Tables = tables
+
+	return nil
+}
+
+func (d *Mongodb) listFields(collection *mongo.Collection) ([]*schema.Column, error) {
+	pipeline := []bson.D{{{Key: "$sample", Value: bson.D{{Key: "size", Value: d.sampleSize}}}}}
+	cursor, err := collection.Aggregate(d.ctx, pipeline)
+	if err != nil {
+		return nil, err
+	}
+	columns := []*schema.Column{}
+	total := 0.0
+	occurrences := map[string]float64{}
+	for cursor.Next(d.ctx) {
+		var result bson.D
+		if err := cursor.Decode(&result); err != nil {
+			return columns, err
+		}
+		total += 1
+		for key, value := range result.Map() {
+			var valueType string
+			switch value.(type) {
+			case string:
+				valueType = "string"
+			case int64:
+				valueType = "int64"
+			case primitive.D:
+				valueType = "document"
+			case primitive.DateTime:
+				valueType = "datetime"
+			case primitive.A:
+				valueType = "array"
+			case primitive.ObjectID:
+				valueType = "objectId"
+			default:
+				valueType = fmt.Sprintf("%T", value)
+			}
+			column := &schema.Column{
+				Name:     key,
+				Type:     valueType,
+				Nullable: false,
+			}
+			if val, ok := occurrences[key]; ok {
+				occurrences[key] = val + 1
+			} else {
+				occurrences[key] = 1
+			}
+			if !columnInColumns(column, columns) {
+				columns = append(columns, column)
+			}
+		}
+	}
+	for _, col := range columns {
+		if stat, ok := occurrences[col.Name]; ok {
+			col.Comment = fmt.Sprintf("Occurrences: %d; Percents: %.1f", int(stat), stat/total*100)
```

Comment:
I think the comments should contain user-defined strings.

For example, how about adding an extra definitions column like mysql driver?

https://github.com/k1LoW/tbls/blob/main/sample/mysql/comments.md

Reply:
fixed in [5c29f60](https://github.com/k1LoW/tbls/pull/339/commits/5c29f601a998934be81a3bcf2990e5d4bc319fe0) 
I am not sure that my change in [xlsx](https://github.com/k1LoW/tbls/pull/339/commits/5c29f601a998934be81a3bcf2990e5d4bc319fe0#diff-0959dd62d685c0908a9fe9a165a45447ced528cdd108c99ae04abdba52f5fdecR152) is correct. Could you please help me with reviewing it? thanks in advance